agent/attachment.go

```diff
@@ -0,0 +1,103 @@
+package agent
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/identity"
+	"github.com/docker/swarmkit/log"
+	"golang.org/x/net/context"
+)
+
+// Listener allow to receive notifications about non executor attachment objects.
+type Listener interface {
+	// Notify notifies the listener about the updates over the executor attachment objects.
+	Notify([]*api.ExecutorAttachment)
+}
+
+// ExecutorAttachmentManager provides control over executor attachments on this node.
+type ExecutorAttachmentManager interface {
```

Comment:
I was not sure how would docker engine code interact with the agent in order to request the attachment allocation and retrieve the attachments.

My understanding is that for the task mgmt it is all inside the agent and directive comes from agent to docker engine. While in this case the trigger comes from outside the agent.

So I thought of agent providing an attachment management point, through which engine code can do init operations and query info about attachments. 
I feel interfaces work very well for this. So that if in future a new object (volumes?) is allowed to be managed, a new volume management point interface can be provided to the agent's clients.

agent or any other internal structure would be the implementer, of each or of all the management points, but this is not relevant to the interface's user.